templates/common/infra/bicep/core/host/container-app.bicep

```diff
@@ -5,13 +5,19 @@ param tags object = {}
 param containerAppsEnvironmentName string
 param containerName string = 'main'
 param containerRegistryName string
+param secrets array = []
 param env array = []
 param external bool = true
 param imageName string
-param keyVaultName string = ''
-param managedIdentity bool = !empty(keyVaultName)
 param targetPort int = 80
 
+@description('Managed identity type')
+@allowed([ 'SystemAssigned', 'UserAssigned' ])
+param managedIdentityType string = 'SystemAssigned'
+
+@description('User assigned identity name')
+param userAssignedIdentityName string = ''
```

Comment:
@jongio Curious about your thoughts here. I think this is the first time we're introducing user identities. I can see this adding a lot of flexibility (we now have an indirection), and some degree of complexity. Not sure what is the best design language.

Reply:
I think we should allow them to specify, but also create if they don't specify existing.